shop/src/main/java/ro/msg/learning/shop/Services/ProductService.java

```diff
@@ -0,0 +1,65 @@
+package ro.msg.learning.shop.Services;
+
+import javassist.NotFoundException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import ro.msg.learning.shop.DTOs.ProductDTO;
+import ro.msg.learning.shop.Entities.Product;
+import ro.msg.learning.shop.Entities.ProductCategory;
+import ro.msg.learning.shop.Repositories.ProductRepository;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class ProductService {
+    @Autowired
+    private ProductRepository productRepository;
+
+    public ProductDTO create(Product product){
```

Comment:
You should have as parameter a ProductDTO instead of a Product. A good practice is to have access to the Domain Layer (our entities) only through the Services. If we use the Data Model also in Controllers (to call the services) we will have multiple layers invoking the Domain Model. So please use the Dtos to communicate between Controllers and Services.